environment.js

```diff
@@ -0,0 +1,26 @@
+var path = require('path');
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var glob = require('glob');
+
+export.ensureDirectory = function(directoryName) {
+  if (!fs.existsSync(directoryName)) {
+    mkdirp.sync(directoryName);
+  }
+};
+
+export.loadSpecifications = function(mocha, directory) {
+   function addSpecs(error, files) {
```

Comment:
Yeah, originally the function was anonymous and then it definitely was difficult to read. The functionality divides into two parts: the call back and the directory read.

The key part of this function is actually the callback that loads the specifications into the runner and the glob call is the secondary part that is simply generating input.

By calling out the behaviour first it makes it easier to understand the intent of the function rather than emphasising the glob by putting the behaviour inside the glob call.